bee-network/src/swarm/builder.rs

```diff
@@ -0,0 +1,76 @@
+// Copyright 2020-2021 IOTA Stiftung
+// SPDX-License-Identifier: Apache-2.0
+
+use super::behavior::SwarmBehavior;
+
+use crate::{network::meta::Origin, service::event::InternalEventSender};
+
+use libp2p::{
+    core::{
+        connection::ConnectionLimits,
+        upgrade::{self, SelectUpgrade},
+    },
+    dns, identity, mplex, noise,
+    swarm::SwarmBuilder,
+    tcp, yamux, Swarm, Transport,
+};
+
+use std::{io, time::Duration};
+
+const MAX_CONNECTIONS_PER_PEER: u32 = 1;
+const DEFAULT_CONNECTION_TIMEOUT_SECS: u64 = 10;
+
+// TODO: should we return errors or just panic?
```

Comment:
Yes, you're right. When I wrote this I was thinking from the perspective of the node. If the OS didn't provide us an address to bind to, then why returning an Error instead of simply panicking, because there is noooo way to salvage that situation during runtime. But that's something `bee-node` has to do, not `bee-network`. So you're abosutely right abou that.

Reply:
Should now be panic-free. (not the test code oc)